Remove SQL suffix from form processor models

millerdev opened the pull request:
Review by commit? Should I split this out into a separate PR for each commit?

The underlying SQL tables are not being renamed, although they could be if we thought it was worthwhile.

This PR may conflict with other open PRs, and new occurrences of the old model names will need to be updated (hopefully that would cause tests to fail so it should be obvious).

## Safety Assurance

### Safety story
Renaming all occurrences of model classes in the codebase.

### Automated test coverage

There are many tests that touch these classes.

### QA Plan

No QA.

### Migrations

Some migration files were renamed, but no new migrations were added.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

snopoke commented on corehq/form_processor/models.py:
```diff
@@ -51,7 +51,7 @@
 XFormOperationSQL_DB_TABLE = 'form_processor_xformoperationsql'
 
 CommCareCaseSQL_DB_TABLE = 'form_processor_commcarecasesql'
```
why not also rename these other constants?

millerdev replied:
Thanks, did this during the migration rework.

millerdev commented:
I think I did the migrations the wrong way. Reworking, which will probably involve a force-push...

millerdev committed:
XFormInstanceSQL -> XFormInstance

millerdev committed:
Lint

millerdev committed:
XFormOperationSQL -> XFormOperation

millerdev committed:
CommCareCaseSQL -> CommCareCase

millerdev committed:
CaseAttachmentSQL -> CaseAttachment

millerdev committed:
CommCareCaseIndexSQL -> CommCareCaseIndex

millerdev committed:
Squash model rename migrations

millerdev committed:
Fix migrations.lock

millerdev commented:
Do we pickle any of these classes anywhere? If yes, I should add aliases so they can be imported for unpickling.

snopoke commented:
> Do we pickle any of these classes anywhere? If yes, I should add aliases so they can be imported for unpickling.

Not that I'm aware of. But we do serialize them in the `dump_load` app. It might be an issue for any data that was dumped prior to this change.

millerdev committed:
Add dump/load support for old class names

millerdev committed:
Merge remote-tracking branch 'origin/master' into dm/rename-sql-models

millerdev committed:
Seek to beginning of output stream

`_parse_dump_output` was recently changed to use `readlines` rather than `getvalue`, which requires the stream pointer to be reset.
See commit ca55a89d9405664b4afa559ab80a5db1d9874938
